src/styles/main.scss

```diff
@@ -29,72 +45,180 @@ body {
     }
 
-    &.selected {
-      background: $url-blue;
+    .result--title {
+      @include overflow-at(65%);
+    }
 
-      .result-copy {
-        color: $white;
-      }
+    .result--url {
+      @include overflow-at(30%);
+      color: $color--blue;
+    }
 
-      .result-url {
-        color: $white;
-        overflow-x: hidden;
+    .result--action {
+      @include overflow-at(30%);
+      color: $color--green;
+    }
+
+    &:hover {
+      background: $color--light-grey;
+      border-color: darken($color--light-grey, 3%);
+    }
+
+    &.result--selected {
+      background: $color--blue;
+      border-color: darken($color--blue, 3%);
+
+      .result--title,
+      .result--action,
+      .result--url,
+      .result--segment + .result--segment:before,
+      .little-machine--segment {
+        color: $color--white;
       }
 
       .icon-search-remote {
-        background-position: 0 (-1 * $icon-unit);
+        background-position: 0 (-1 * $size--icon);
       }
 
       .icon-bookmark {
-        background-position: (-1 * $icon-unit) (-1 * $icon-unit);
+        background-position: (-1 * $size--icon) (-1 * $size--icon);
       }
 
       .icon-search-local {
-        background-position: (-2 * $icon-unit) (-1 * $icon-unit);
+        background-position: (-2 * $size--icon) (-1 * $size--icon);
       }
     }
   }
+
+  .result--segment--little-machine {
+    flex-direction: column;
+    min-height: $size--preview;
+    padding-left: ($size--preview + $size--preview--spacing);
+    position: relative;
+
+    .little-machine--segment {
+      margin: 0;
+    }
+
+    .little-machine--image {
+      border-radius: 5px;
+      box-shadow: 0 0 1px rgba(0, 0, 0, 0.25) inset;
+      left: 0;
+      line-height: 0;
+      overflow: hidden;
+      position: absolute;
+      top: 0;
+
+      img {
+        height: $size--preview;
+        position: relative;
+        width: $size--preview;
+        z-index: -1;
+      }
+    }
+
+    .little-machine--title {
+      color: $color--black;
+      font-size: 14px;
+    }
+
+    .little-machine--abstract,
+    .little-machine--url {
+      color: $color--dark-grey;
+      font-size: 13px;
+      margin-top: 4px;
+    }
+
+    .little-machine--url {
+      color: $color--blue;
+    }
+  }
+}
+
+
+[data-mozaction="switchtab"] .result--segment[data-mozaction~="switchtab"],
+[data-mozaction="other"] .result--segment[data-mozaction~="other"] {
+  display: inline-block;
+}
+
+.icon {
+  background-repeat: no-repeat;
+  background-size: $size--icon;
+}
+
+.icon--action {
+  [data-mozaction="switchtab"] & {
+    background-image: url(../images/icons/tab.svg);
+  }
+
+  [data-mozaction="switchtab"].result--selected & {
+    background-image: url(../images/icons/tab-selected.svg);
+  }
+}
+
+.icon--recommendation {
+  background-image: url(../images/icons/recommendation.svg);
+
+  .result--selected & {
+    background-image: url(../images/icons/recommendation-selected.svg);
+  }
 }
 
 .icon {
```

Comment:
Ah, for reasons that will never be clear to me, the `sass-lint` CLI returns zero output unless you add the `--verbose` flag. Seems weird, but OK.

So I added this to package.json:
```js
"sass-lint": "sass-lint src/styles/**/*.scss --verbose",
```

Create the following `.sass-lint.yml` file:
```yml
rules:
  force-attribute-nesting: 0
  force-element-nesting: 0
  force-pseudo-nesting: 0
  nesting-depth: [2, max-depth: 3]
  no-ids: 0
  no-qualifying-elements: 0
  quotes: 0
```

Run this code from my Terminal:
```sh
npm run sass-lint
```

And then get this output (after I tweak a bit of Sass locally):
```sh
✗ npm run sass-lint

> universal-search-content@0.0.1 sass-lint /Users/pdehaan/dev/tmp/del/universal-search-content
> sass-lint src/styles/**/*.scss --verbose


src/styles/main.scss
   36:3   warning  Expected `-moz-user-select`, found `box-sizing`                              property-sort-order
   37:3   warning  Expected `box-sizing`, found `-moz-user-select`                              property-sort-order
   37:3   warning  Vendor prefixes should not be used                                           no-vendor-prefixes
   73:17  warning  Indentation of 16, expected 4                                                indentation
   74:17  warning  Indentation of 16, expected 4                                                indentation
  158:27  warning  Color functions such as 'rgba' should only be used in variable declarations  no-color-literals
  220:1   warning  Rule `.icon` should be merged with the rule on line 197                      no-mergeable-selectors
  252:15  warning  Color literals such as 'eee' should only be used in variable declarations    no-color-literals
  253:28  warning  Color literals such as 'ccc' should only be used in variable declarations    no-color-literals
  257:12  warning  Color literals such as '555' should only be used in variable declarations    no-color-literals
  269:25  warning  Color literals such as 'ccc' should only be used in variable declarations    no-color-literals
  274:21  warning  Color literals such as 'ddd' should only be used in variable declarations    no-color-literals

✖ 12 problems (0 errors, 12 warnings)
```

Anyways, I'm _waaaayyy_ off topic here, so I'll file a separate bug or two and we can discuss elsewhere if this is useful.